neutrino.go

```diff
@@ -945,6 +957,39 @@ func NewChainService(cfg Config) (*ChainService, error) {
 		permanentPeers = cfg.AddPeers
 	}
 
+	// Adding rest peers to chainservice if spesified in config
+
+	restPeers := cfg.RestPeers
+	if len(restPeers) != 0 {
+		s.client, err = NewHTTPClient()
+		if err != nil {
+			log.Debugf("error setting up HTTPClient: %w", err)
+		}
+		// Iterating thought restpeer defined in the config
+		// and checking if they are reachable.
+		for _, restAddr := range restPeers {
+			_, err := url.Parse(restAddr)
+			if err != nil {
+				log.Debugf("error unable to parse address: %w", err)
+				break
+			}
+
+			res, err := s.client.Get(restAddr)
```

Comment:
The server is not supposed to answer to GET queries here, even if one specific implementation returns OK, it's wrong to check if a server is ok at the init stage